cpp/src/fil/treelite_import.cu

```diff
@@ -771,46 +692,20 @@ void from_treelite(const raft::handle_t& handle,
     }
   }
 
-  forest_params_t params;
-  cat_sets_owner cat_sets;
   switch (storage_type) {
     case storage_type_t::DENSE: {
-      std::vector<dense_node> nodes;
-      std::vector<float> vector_leaf;
-      tl2fil_dense(&nodes, &params, model, tl_params, &cat_sets, &vector_leaf);
-      init_dense(handle, pforest, cat_sets.accessor(), vector_leaf, nodes.data(), &params);
-      // sync is necessary as nodes is used in init_dense(),
-      // but destructed at the end of this function
-      CUDA_CHECK(cudaStreamSynchronize(handle.get_stream()));
-      if (tl_params->pforest_shape_str) {
-        *tl_params->pforest_shape_str = sprintf_shape(model, storage_type, nodes, {}, cat_sets);
-      }
+      tl2fil_t<dense_node, threshold_t, leaf_t> tl2fil(model, *tl_params);
```

Comment:
These two lines are repeated 3 times, for each node type. Consider adding a static method that does it to `tl2fil_t` instead, and call it here.

Reply:
such a change adds many more lines than it saves (22 vs 10), and the removable repeated part
```
); 
tl2fil.init_object(); 
tl2fil.init_forest(
```
is hard to read wrong or write wrong upon altering.